### PR TITLE
[tiny-process-library] Bump tiny-process-library to 2.0.4

### PR DESCRIPTION
**Pull request by traversaro**
**Describe the pull request**

- What does your PR fix? This PR updates the tiny-process-library to the 2.0.3

- Which triplets are supported/not supported? Have you updated the CI baseline? The baseline should not be affected.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)? Yes

### CODE REVIEW

**Comment by PhoebeHui** [comment]
@traversaro, thanks for the PR!

Here is CI testing results, could you please take a look?

x86-windows/x64-windows/arm64-windows/
```
LINK : fatal error LNK1104: cannot open file 'tiny-process-library.lib'
```
[failure logs for x64-windows (1).zip](https://github.com/microsoft/vcpkg/files/5587098/failure.logs.for.x64-windows.1.zip)


x64-windows-static/x64-linux
```
-- Configuring x64-windows-static
-- Building x64-windows-static-dbg
-- Building x64-windows-static-rel
CMake Error at scripts/cmake/vcpkg_fixup_cmake_targets.cmake:73 (message):
  'D:/packages/tiny-process-library_x64-windows-static/debug/lib/cmake/tiny-process-library'
  does not exist.
Call Stack (most recent call first):
  ports/tiny-process-library/portfile.cmake:21 (vcpkg_fixup_cmake_targets)
  scripts/ports.cmake:136 (include)

Error: Building package tiny-process-library:x64-windows-static failed with: BUILD_FAILED
```

**Comment by traversaro** [comment]
> @traversaro, thanks for the PR!

Thanks for the comment @PhoebeHui  ! I fixed all the problems, and now the port should work fine, I am not sure what happened to the CI jobs that staled.

File: ports/tiny-process-library/portfile.cmake [hunk 1]
```diff
@@ -2,27 +2,26 @@ vcpkg_from_gitlab(
...
-    PATCHES
-        fix-cmake.patch
 )
 
 vcpkg_configure_cmake(
     SOURCE_PATH ${SOURCE_PATH}
     PREFER_NINJA
     OPTIONS
         -DBUILD_TESTING=OFF
+        -DCMAKE_WINDOWS_EXPORT_ALL_SYMBOLS=ON
```
Comment (PhoebeHui) [inline ports/tiny-process-library/portfile.cmake:15]: Is the author of the library using it? if not, we should not add this in vcpkg. See https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#do-not-add-cmake_windows_export_all_symbols

**Review by PhoebeHui — CHANGES_REQUESTED** [review]

File: ports/tiny-process-library/portfile.cmake (hunk 1 above)
Comment (traversaro) [reply ports/tiny-process-library/portfile.cmake:15]: `tiny-process-library` is a small library with a few function and no templates, so `CMAKE_WINDOWS_EXPORT_ALL_SYMBOLS` works as intended. It was already used in the patch that this PR removes. Furthermore, it was also added upstream in https://gitlab.com/eidheim/tiny-process-library/-/merge_requests/39, but it is setfor an error **after** the `add_library` call, so it does not work as intended. Passing it for a command line is just a way to solve this problem before it is solved upstream (like a patch, but more future-proof).

Comment (PhoebeHui) [reply ports/tiny-process-library/portfile.cmake:15]: @strega-nil, could you help confirm?

Comment (strega-nil) [reply ports/tiny-process-library/portfile.cmake:15]: Yeah, LGTM

**Comment by traversaro** [comment]
fyi @j-rivero, if you want to include `tiny-process-library` in the Gazebo vcpkg dependencies for compiling the Gazebo , after this PR  the version that exports the correct CMake imported target is available in vpckg.

**Comment by traversaro** [comment]
And thanks a lot @PhoebeHui and @strega-nil  for reviewing and merging this!

**Comment by j-rivero** [comment]
> fyi @j-rivero, if you want to include `tiny-process-library` in the Gazebo vcpkg dependencies for compiling the Gazebo main command on Windows, after this PR the version that exports the correct CMake imported target is available in vpckg.

Thanks Silvio. Gazebo classic is not using dependencies from vcpkg in our CI, something I will need to fix or change at some point.

**Pull request merged**